tests/unit/components/playback/controls/StimulationStudioWidget.spec.js

```diff
@@ -0,0 +1,161 @@
+import { mount } from "@vue/test-utils";
+import StimulationStudioWidget from "@/components/playback/controls/StimulationStudioWidget.vue";
+import { StimulationStudioWidget as DistComponentToTest } from "@/dist/mantarray.common";
+import Vuex from "vuex";
+import { createLocalVue } from "@vue/test-utils";
+
+let wrapper = null;
+
+const localVue = createLocalVue();
+localVue.use(Vuex);
+let NuxtStore;
+let store;
+
+describe("StimulationStudioWidget.vue", () => {
+  beforeAll(async () => {
+    // note the store will mutate across tests, so make sure to re-create it in beforeEach
+    const storePath = `${process.env.buildDir}/store.js`;
+    NuxtStore = await import(storePath);
+  });
+
+  beforeEach(async () => {
+    store = await NuxtStore.createStore();
+  });
+
+  afterEach(() => wrapper.destroy());
+  test("When mounting StimulationStudioWidget from the build dist file, Then it loads successfully", async () => {
+    const propsData = {
+      protocol_code: [],
+    };
+    wrapper = mount(DistComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const well = wrapper.findAll("circle");
+    expect(well).toHaveLength(24);
+  });
+  test("Given the StimulationStudioWidget is in the range of 0 - 25, When mounted successfully, Then verify that Stimulation Plate Well is teal color and Alphabets A-X is visible", async () => {
+    const protocol_list = Array.from(Array(26).keys());
+
+    const propsData = {
+      protocol_code: protocol_list,
+    };
+    wrapper = mount(StimulationStudioWidget, {
+      propsData,
+      store,
+      localVue,
+    });
+    const well = wrapper.findAll("circle");
+    expect(well.at(0).attributes("fill")).toStrictEqual("#19AC8A");
+  });
+  test("Given the StimulationStudioWidget is in the range of 0 - 23, When mounted successfully, Then verify that Stimulation Plate Well is teal color and Alphabets A-X is visible", async () => {
+    const protocol_list = Array.from(Array(24).keys());
+
+    const propsData = {
+      protocol_code: protocol_list,
+    };
+    wrapper = mount(StimulationStudioWidget, {
+      propsData,
+      store,
+      localVue,
+    });
+    const well = wrapper.findAll("circle");
+    const protocol_name = wrapper.findAll(
+      ".span__simulationstudio-plate-well-protocol-location"
+    );
+    expect(well.at(0).attributes("fill")).toStrictEqual("#19AC8A");
+    expect(protocol_name.at(0).text()).toStrictEqual("A");
+    expect(well.at(23).attributes("fill")).toStrictEqual("#19AC8A");
+    expect(protocol_name.at(23).text()).toStrictEqual("X");
+  });
+  test("Given the StimulationStudioWidget is in the range of 24 - 48, When mounted successfully, Then verify that Stimulation Plate Well is teal color(at 0) and blue color(at 23) and Alphabets Y-V is visible", async () => {
```

Comment:
As you are writing your tests before writing the next piece of code, one way to help make sure that you are following the TDD guidance of only writing the simplest piece of code to make the current failing test pass is to test the edge cases. For example here, you expect a color change to happen between index 25 and 26 (and also the alphabet to restart), so those would be relevant places to make assertions

Reply:
Not essentially required VRT covers a good amount of all scenarios.